modelforge/potential/ani.py

```diff
@@ -657,113 +666,59 @@ def compute_properties(self, data: AniNeuralNetworkData) -> Dict[str, torch.Tens
-    angular_dist_divisions : int
-        The number of divisions for the angular distance.
-    angle_sections : int
-        The number of angle sections to use.
-    activation_function_parameter : Dict
-        Dict that contains keys: activation_function_name [str], activation_function_arguments [Dict],
-        and activation_function [Type[torch.nn.Module]].
-    postprocessing_parameter : Dict[str, Dict[str, bool]]
-        Configuration for postprocessing parameters.
-    dataset_statistic : Optional[Dict[str, float]], optional
-        Statistics of the dataset, by default None.
-    potential_seed : Optional[int], optional
-        Seed for the random number generator, default None.
-    """
-
-    def __init__(
-        self,
-        maximum_interaction_radius: Union[unit.Quantity, str],
-        minimum_interaction_radius: Union[unit.Quantity, str],
-        number_of_radial_basis_functions: int,
-        maximum_interaction_radius_for_angular_features: Union[unit.Quantity, str],
-        minimum_interaction_radius_for_angular_features: Union[unit.Quantity, str],
-        angular_dist_divisions: int,
-        angle_sections: int,
-        activation_function_parameter: Dict,
-        postprocessing_parameter: Dict[str, Dict[str, bool]],
-        predicted_properties: List[Dict[str, str]],
-        dataset_statistic: Optional[Dict[str, float]] = None,
-        potential_seed: Optional[int] = None,
-    ) -> None:
-        from modelforge.utils.units import _convert_str_to_unit
-
-        self.only_unique_pairs = True  # NOTE: need to be set before super().__init__
-
-        super().__init__(
-            dataset_statistic=dataset_statistic,
-            postprocessing_parameter=postprocessing_parameter,
-            maximum_interaction_radius=_convert_str_to_unit(maximum_interaction_radius),
-            potential_seed=potential_seed,
-        )
 
-        activation_function = activation_function_parameter["activation_function"]
-
-        self.core_module = ANI2xCore(
-            maximum_interaction_radius=_convert_str_to_unit(maximum_interaction_radius),
-            minimum_interaction_radius=_convert_str_to_unit(minimum_interaction_radius),
-            number_of_radial_basis_functions=number_of_radial_basis_functions,
-            maximum_interaction_radius_for_angular_features=_convert_str_to_unit(
-                maximum_interaction_radius_for_angular_features
-            ),
-            minimum_interaction_radius_for_angular_features=_convert_str_to_unit(
-                minimum_interaction_radius_for_angular_features
-            ),
-            activation_function=activation_function,
-            angular_dist_divisions=angular_dist_divisions,
-            angle_sections=angle_sections,
-            predicted_properties=predicted_properties,
+    def _aggregate_results(
+        self, outputs: Dict[str, torch.Tensor]
+    ) -> Dict[str, torch.Tensor]:
+        per_atom_prediction = outputs.pop("per_atom_prediction")
```

Comment:
I'm a little confused what is going on in this function.  Is this just basically just changing the hierarchy of the dict?  Can you add in some comments about what this does.

Reply:
Updated the docstring